### DISCUSSION TITLE
Keycloak - apache - tomcat - spring boot adapter-proxy

### DISCUSSION THREAD

**Discussion by applicationuser** [keycloak/keycloak#12802] [Q&A]
Hello,
Need you help, 

We have angular web application with spring boot  REST services deployed on tomcat, Using Keycloak for IAM
Apache web server setup for external application access .
Self signed certificate installed on apache for testing. (not production environment)
No https required between apache and keycloak.   so proxy configured as 'edge' mode.

Here is my environment setup

````
OS: Windows 
All applications setup on single server
Keycloak
	Version: 17.0.1
	run params:	kc.bat start --http-port=9090 --proxy=edge --hostname=<servername> 
        --hostname-strict-https=false     --http-enabled=true --auto-build
Apache 	Webserver (Apache Haus)
	Version: 2.4
	httpd-ahssl.conf
<VirtualHost *:443>
  SSLEngine on
  ServerName <servername>:443
  ServerAlias <servername>
  SSLProtocol all
  SSLCipherSuite HIGH:3DES:!aNULL:!MD5:!SEED:!IDEA
  SSLCertificateFile "${SRVROOT}/conf/ssl/server.crt"
  SSLCertificateKeyFile "${SRVROOT}/conf/ssl/server.key"
  SSLCertificateChainFile "${SRVROOT}/conf/ssl/server.crt"
 
	ProxyPreserveHost On	
    ProxyRequests Off
	RequestHeader add "X-Forwarded-For" "X-Forwarded-Server"
		
	RequestHeader set x-ssl-client-cert "${SRVROOT}/conf/ssl/server.crt"
	
	RequestHeader set SSL_CLIENT_CERT "%{SSL_CLIENT_CERT}s"
	RequestHeader set SSL_CIPHER "%{SSL_CIPHER}s"
	RequestHeader set SSL_SESSION_ID "%{SSL_SESSION_ID}s"
	RequestHeader set SSL_CIPHER_USEKEYSIZE "%{SSL_CIPHER_USEKEYSIZE}s"
  
	SSLProxyEngine On
	SSLProxyCheckPeerCN on
	SSLProxyCheckPeerExpire on
	
	RequestHeader set X-Forwarded-Proto "https"
	RequestHeader set X-Forwarded-Port "443"
	
	ProxyPass /realms/ http://localhost:9090/realms/
    ProxyPassReverse /realms/ http://localhost:9090/realms/
		
	ProxyPass /js/ http://localhost:9090/js/
    ProxyPassReverse /js/ http://localhost:9090/js/
	
	ProxyPass /resources/ http://localhost:9090/resources/
    ProxyPassReverse /resources/ http://localhost:9090/resources/
				
	ProxyPass /api http://localhost:8080/<SpringBootServices>
    ProxyPassReverse /api http://localhost:8080/<<SpringBootServices>
	
	ProxyPass /AngularWeb/ http://localhost:8080/AngularWeb/
    ProxyPassReverse /AngularWeb/ http://localhost:8080/AngularWeb/
	
</virtualhost>

Tomcat Server 
	Version: 9.0
        port: 8080
Backend Services with Spring Boot - keycloak-spring-security-adapter
	Version: 17.0.1
	keycloak.auth-server-url=http://localhost:9090
Angular- auth
	angular-auth-oidc-client- version 13.1.0
````
Access to angular application with apache and user login with keycloak works fine.  Access token generated and sent as header request to back end services.
On accessing back end REST services , receiving auth error message : 

`**error="invalid_token", error_description="Invalid token issuer. Expected 'http://<servername>/realms/<realmname>', but was 'https://<servername>/realms/<realmname>**'"`

Keycloak admin console access not required from external url. only internal access within server.

Not able to access admin console with URL http://localhost:9090/admin/master/console , white page with frequent screen refresh.
`
**Error : 404 not found for url : http://<Servername>/realms/master/protocol/openid-connect/3p-cookies/step1.html?version=w3fck**`

**Comment by Padmabishek** [comment]
Any solution for the above issue will be appreciated . I am facing the same problem
